From 120b4a9cf6deb45f040d36412b0aba9cb2fa4798 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 24 Aug 2011 09:51:18 +0200
Subject: [PATCH] e2image: fix invalid lseek error detection

In flush_l2_cache() we are using ext2fs_llseek() however we do not
properly detect the error code returned from the function, because we
are assigning it into ULL variable, hence we will not see negative
values.

Fix this by changing the type of the variable to ext2_loff_t which is
signed and hence will store negative values.

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 misc/e2image.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/misc/e2image.c b/misc/e2image.c
index 9738a36..3f60315 100644
--- a/misc/e2image.c
+++ b/misc/e2image.c
@@ -762,7 +762,8 @@ static void put_used_table(struct ext2_qcow2_image *img,
 
 static void flush_l2_cache(struct ext2_qcow2_image *image)
 {
-	blk64_t offset, seek = 0;
+	blk64_t seek = 0;
+	ext2_loff_t offset;
 	struct ext2_qcow2_l2_cache *cache = image->l2_cache;
 	struct ext2_qcow2_l2_table *table = cache->used_head;
 	int fd = image->fd;
-- 
1.7.4.4

